### PR TITLE
macOS toolbar improvements

### PR DESCRIPTION
**Pull request by melimion**
### Description of Change ###
Fixes toolbar features that are not implemented or working incorrectly for macOS:

- implements BackButtonTitle
- implements handling of ToolbarItem icon changes
- fixes toolbar item width when icon/text changes
- fixes toolbar item synchronization with current page

### API Changes ###
 None

### Platforms Affected ### 
- macOS

### Behavioral/Visual Changes ###

- BackButtonTitle works
- Toolbar items width and position is correct
- Toolbar items always match to current page's toolbar items

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###
Control gallery pages:
- ToolbarItems Gallery
- NavigationBar Gallery - Legacy

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Platform.MacOS/NativeToolbarTracker.cs [hunk 1]
```diff
@@ -422,18 +410,73 @@ void UpdateGroup(NativeToolbarGroup group, IList<ToolbarItem> toolbarItems, doub
...
+				{
+					buttonWidth = child.FittingSize.Width + 10;
+				}
+				child.Frame = new CGRect(currentX + i * group.ItemSpacing, 0, buttonWidth, ToolbarItemHeight);
+				totalWidth += child.Frame.Width;
+				currentX += buttonWidth;
+			}
+			view.Frame = new CGRect(0, 0, totalWidth + (group.ItemSpacing * (view.Subviews.Length - 1)), ToolbarItemHeight);
+		}
+		void SetButtonTitle(NSButton button, string text)
```
Comment (jsuarezruiz) [inline Xamarin.Forms.Platform.MacOS/NativeToolbarTracker.cs]: New line between the methods.

File: Xamarin.Forms.Platform.MacOS/NativeToolbarTracker.cs [hunk 2]
```diff
@@ -422,18 +410,73 @@ void UpdateGroup(NativeToolbarGroup group, IList<ToolbarItem> toolbarItems, doub
...
+			nfloat totalWidth = 0;
+			var currentX = 0.0;
+			var view = group.Group.View;
+			for (var i = 0; i < view.Subviews.Length; i++)
+			{
+				var child = view.Subviews[i];
+				var buttonWidth = group.MinItemWidth;
+				if (child.FittingSize.Width > group.MinItemWidth)
+				{
+					buttonWidth = child.FittingSize.Width + 10;
```
Comment (jsuarezruiz) [inline Xamarin.Forms.Platform.MacOS/NativeToolbarTracker.cs]: I have tried using directly `child.FittingSize.Width` and the result seems to be correct. Adding 10 to the width slightly increases the spacing, is that the goal?

Comment (melimion) [reply Xamarin.Forms.Platform.MacOS/NativeToolbarTracker.cs]: Hmm, this line was taken from existing implementation. It adds some horizontal padding when button content is long.

Padding for toolbar items with icon only is 9=44-25 (ToolbarItemWidth-ToolbarItemHeight)
I think that 10 added here to match this padding for long items.

Comment (melimion) [reply Xamarin.Forms.Platform.MacOS/NativeToolbarTracker.cs]: @jsuarezruiz I've added toolbar item with long text to ControlGallery page

Comment (jsuarezruiz) [reply Xamarin.Forms.Platform.MacOS/NativeToolbarTracker.cs]: Ok, sounds good.

**Comment by knocte** [comment]
@samhouts can we have more reviewers? there's already one approval; thanks

**Comment by jfversluis** [comment]
@melimion would you be able to rebase it on top of the latest master? Thanks!

**Comment by melimion** [comment]
@jfversluis done

**Pull request merged**